app/controllers/RiderStageConnectionController.java

```diff
@@ -19,22 +21,20 @@
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executors;
 
+import static play.libs.Json.toJson;
+
 public class RiderStageConnectionController extends Controller {
-    private final StageRepository stageRepository;
-    private final RiderRepository riderRepository;
     private final RiderStageConnectionRepository riderStageConnectionRepository;
 
     @Inject
-    public RiderStageConnectionController(StageRepository stageRepository, RiderRepository riderRepository, RiderStageConnectionRepository riderStageConnectionRepository) {
-        this.stageRepository = stageRepository;
-        this.riderRepository = riderRepository;
+    public RiderStageConnectionController(RiderStageConnectionRepository riderStageConnectionRepository) {
         this.riderStageConnectionRepository = riderStageConnectionRepository;
     }
 
-    /*
-    public CompletionStage<Result> getRiderStageConnections() {
-        return riderStageConnectionRepository.getAllRiderStageConnections().thenApplyAsync(riderStageConnections -> {
-            return ok(riderStageConnections);
+
+    public CompletionStage<Result> getRiderStageConnections(long stageId) {
+        return riderStageConnectionRepository.getAllRiderStageConnections(stageId).thenApplyAsync(riderStageConnections -> {
+            return ok(toJson(riderStageConnections));
         }).exceptionally(ex -> {
             Result res = null;
             switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)